### PR TITLE
Task3 yellow

### PR DESCRIPTION
**Pull request by TillSchlemmermeier**
3.1 Backend: Units mit Order für Materials erweitern und Metadaten hinzufügen (Titel,Autor und Erstelldatum)

### CODE REVIEW

File: app/controllers/units_controller.rb [hunk 1]
```diff
@@ -16,8 +17,7 @@ def show
   # POST /units.json
   def create
     @unit = Unit.new(unit_params)
-
-    if @unit.save
+    if @unit.save && @unit.errors.empty? && @unit.update(creator: @current_user['name'].to_s)
```
Comment (malachewhiz) [inline app/controllers/units_controller.rb]: we should try to only do `@unit.save` and just test if it returns true and false, without further test.
We can do it easily here:
```ruby
@unit = Unit.new(unit_params)
@unit.creator = @current_user['name']
if @unit.save
```
And please edit the Content's property name to `creator_name`, if it's just the name (a String) it is not mistaken as for example a material's creator (a Person)

File: app/models/unit.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@ class Unit
   property :title, type: String
   property :updated_at, type: DateTime
   property :created_at, type: DateTime
+  property :creator, type: String
```
Comment (malachewhiz) [inline app/models/unit.rb]: Please change to
```ruby
property :creator_name, type: String
```

**Comment by TillSchlemmermeier** [comment]
changes done

**Comment by malachewhiz** [comment]
Ok, after discussing this topic with @patous we came to this conclusion:

Saving the creator as a string property on the unit is insufficient. The trello card wasn't clear on this behalf. You should save an edge to the Person like it's done [here.](https://github.com/HamburgOOU/neo-rails/blob/master/app/models/material.rb#L7)

Furthermore, the unit specs for testing the ordering are way to few, you basically test if an empty array is an empty array. There should be specs which add materials to a unit in a specific order and test if the order will be persisted.

And please run rubocop before issuing a new code review.

**Comment by TillSchlemmermeier** [comment]
works now

File: app/controllers/units_controller.rb [hunk 3]
```diff
@@ -27,7 +33,14 @@ def create
   # PATCH/PUT /units/1
   # PATCH/PUT /units/1.json
   def update
-    if @unit.update(unit_params)
+    @unit.update(unit_params)
+    if @unit.save
```
Comment (patous) [inline app/controllers/units_controller.rb]: `@unit.update` and `@unit.save` both persist changes in the hash received to the database. If you're commiting changes to the database with `#save` the `#update` is redundant.

**Comment by patous** [comment]
OK to merge, once the issue referred to in code comment is fixed.

**Pull request merged**